src/core/Akka.Remote/DeadlineFailureDetector.cs

```diff
@@ -28,7 +28,7 @@ public DeadlineFailureDetector(TimeSpan acceptableHeartbeatPause, Clock clock =
         {
             _acceptableHeartbeatPause = acceptableHeartbeatPause;
             _acceptableHeartbeatMillis = Convert.ToInt64(acceptableHeartbeatPause.TotalMilliseconds);
-            Guard.Assert(_acceptableHeartbeatPause > TimeSpan.Zero, "acceptable-heartbeat-pause must be greater than zero");
+            if (_acceptableHeartbeatPause <= TimeSpan.Zero) throw new ArgumentException("acceptable-heartbeat-pause must be greater than zero");
```

Comment:
If the value refers to a setting in a config, the type should be `Akka.Configuration.ConfigurationException`
The hyphen'ed name in the message hints that this is the case.